### PR TITLE
hide due date time

### PR DESCRIPTION
**Pull request by banukutlu**
Issue #87

### CODE REVIEW

File: app/helpers/checkouts_helper.rb [hunk 1]
```diff
@@ -12,13 +12,19 @@ def render_checkout_due_date(checkout)
...
+      contents << format_due_date(checkout.recall_due_date)
     end
-    contents << l(checkout.due_date, format: :long)
+    contents << format_due_date(checkout.due_date)
 
     content_tag 'span', contents.join('<br>'), nil, false
   end
 
+  def format_due_date(date)
+    return l(date, format: :long) unless l(date, format: :time_only) == '11:59pm'
```
Comment (cdmo) [inline app/helpers/checkouts_helper.rb]: nifty

File: spec/helpers/checkouts_helper_spec.rb [hunk 2]
```diff
@@ -51,20 +51,40 @@
...
         allow(checkout).to receive_messages(
           recalled?: true,
-          recall_due_date: Time.zone.parse('2019-11-10T23:59:00-05:00')
+          recall_due_date: Time.zone.parse('2019-11-10T22:30:00-05:00')
         )
       end
 
       it 'renders the right html' do
-        expect(content).to include('<span>Recalled<br>November 10, 2019 11:59pm<br>November 14, 2019 11:59pm</span>')
+        expect(content).to include('<span>Recalled<br>November 10, 2019 10:30pm<br>November 14, 2019</span>')
```
Comment (cdmo) [inline spec/helpers/checkouts_helper_spec.rb]: Out of curiosity, I wonder why you are suing `include` here and above you used `have_css` for seemingly the same kinda check. Just kinda random thing? I don't know if one is less expensive than the other tbh.

File: spec/helpers/checkouts_helper_spec.rb [hunk 3]
```diff
@@ -51,20 +51,40 @@
...
+    end
+  end
+
+  describe '#format_due_date' do
+    let(:content) { helper.format_due_date(due_date) }
+
+    context 'when due date time is 11:59pm' do
+      let(:due_date) { Time.zone.parse('2019-11-10T23:59:00-05:00') }
+
+      it 'renders the right html' do
```
Comment (cdmo) [inline spec/helpers/checkouts_helper_spec.rb]:
```suggestion
      it 'does not include the time' do
```

File: spec/helpers/checkouts_helper_spec.rb [hunk 4]
```diff
@@ -51,20 +51,40 @@
...
+        expect(content).to eq('November 10, 2019')
+      end
+    end
+
+    context 'when due date time is not 11:59pm' do
+      let(:due_date) { Time.zone.parse('2019-11-10T22:30:00-05:00') }
+
+      it 'renders the right html' do
+        expect(content).to include('November 10, 2019 10:30pm')
```
Comment (cdmo) [inline spec/helpers/checkouts_helper_spec.rb]:
```suggestion
        expect(content).to eq('November 10, 2019 10:30pm')
```

make it consistent with above spec

File: spec/helpers/checkouts_helper_spec.rb [hunk 5]
```diff
@@ -51,20 +51,40 @@
...
+
+      it 'renders the right html' do
+        expect(content).to eq('November 10, 2019')
+      end
+    end
+
+    context 'when due date time is not 11:59pm' do
+      let(:due_date) { Time.zone.parse('2019-11-10T22:30:00-05:00') }
+
+      it 'renders the right html' do
```
Comment (cdmo) [inline spec/helpers/checkouts_helper_spec.rb]:
```suggestion
      it 'includes the time' do
```
to be more specific

File: spec/helpers/checkouts_helper_spec.rb (hunk 2 above)
Comment (banukutlu) [reply spec/helpers/checkouts_helper_spec.rb]: hmmm no specific reason that I can remember of, I will try it with `have_css`, good point

Comment (cdmo) [reply spec/helpers/checkouts_helper_spec.rb]: yeah idunno, dont feel like you have to, it's fine the way it is - it will definitely work either way.

Comment (banukutlu) [reply spec/helpers/checkouts_helper_spec.rb]: updated, see latest

**Pull request merged**